tests/example-all-fail/expected_results.json

```diff
@@ -1,74 +1,167 @@
  {
    "tests": [
-     { "name"  : "Test 1: stating something",
-       "status": "fail" }
-     ,
-     { "name"  : "Test 2: shouting",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 3: shouting gibberish",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 4: asking a question",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 5: asking a numeric question",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 6: asking gibberish",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 7: talking forcefully",
-       "status": "fail" }
-     ,
-     { "name"  : "Test 8: using acronyms in regular speech",
-       "status": "fail" }
-     ,
-     { "name"  : "Test 9: forceful question",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 10: shouting numbers",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 11: no letters",
-       "status": "fail" }
-     ,
-     { "name"  : "Test 12: question with no letters",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 13: shouting with special characters",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 14: shouting with no exclamation mark",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 15: statement containing question mark",
-       "status": "fail" }
-     ,
-     { "name"  : "Test 16: non-letters with question",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 17: prattling on",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 18: silence",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 19: prolonged silence",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 20: multiple line question",
-       "status": "fail" }
-     ,
-     { "name"  : "Test 21: starting with whitespace",
-       "status": "fail" }
-     ,
-     { "name"  : "Test 22: ending with whitespace",
-       "status": "pass" }
-     ,
-     { "name"  : "Test 23: non-question ending with whitespace",
-       "status": "fail" }
+     {
+       "name"     : "Test 1: stating something",
+       "test_code": "Test 1: stating something",
```

Comment:
Was it intentional to add this key? If so, its contents should actually be different, as the `test_code` property should be the code that is run for this test. In this case, that would be:

```
assert_equal("Whatever.", hey("Tom-ay-to, tom-aaaah-to."), "stating something")
```

If you plan to add this later (which sounds like a good plan), I'd suggest to either remove this field or make its value `null`, as otherwise the student will be confused how things look in the editor.

Reply:
@ErikSchierboom I am not sure if I get this point, but as the expected_results.json is written by the Fortran TesterLib I will have to open another PR for this problem - although I am a bit uncertain what the format of expected_results.json actually is. I have to dig more into that....

Reply:
@pclausen My point is that the `test_code` field's contents are wrong. The `test_code` field is currently a copy of the `name` field, but it should be the actual test code. So something like:

```fortran
assert_equal("Whatever.", hey("Tom-ay-to, tom-aaaah-to."), "stating something")
```

Does that make sense?